File: src/main/scala/org/jglr/phiengine/network/Packet.java
package org.jglr.phiengine.network;

import io.netty.buffer.ByteBuf;

public abstract class Packet {
    private String channel;

    public abstract void write(ByteBuf buffer);

    public abstract void read(ByteBuf buffer);

    public String getChannel() {
        return channel;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }
}

File: src/main/scala/org/jglr/phiengine/network/Packet.scala
package org.jglr.phiengine.network

import org.jglr.phiengine.network.utils.NetworkSerializable

abstract class Packet(val id: Int) extends NetworkSerializable {
  private var channel: String = null
  var answerIndex: Long = -1L
  var packetIndex: Long = -1L

  def getChannel: String = {
    channel
  }

  def setChannel(channel: String) {
    this.channel = channel
  }
}